components/constellation/constellation.rs

```diff
@@ -2623,32 +2623,30 @@ where
     }
 
     fn handle_register_serviceworker(&mut self, scope_things: ScopeThings, scope: ServoUrl) {
-        let origin = scope.origin();
-
-        if let Some(mgr) = self.sw_managers.get(&origin) {
-            let _ = mgr.send(ServiceWorkerMsg::RegisterServiceWorker(scope_things, scope));
-        } else {
-            let (own_sender, receiver) = ipc::channel().expect("Failed to create IPC channel!");
+        // This match is equivalent to Entry.or_insert_with but allows for early return.
+        let _ = match self.sw_managers.entry(scope.origin()) {
```

Comment:
Ok, how about assigning to something like `sw_sender`, and then using that to send the message?